consensus/module.go

```diff
@@ -58,7 +65,18 @@ type consensusModule struct {
 	MaxBlockBytes uint64
 }
 
-func Create(cfg *genesis.Config, genesis *genesis.GenesisState) (modules.ConsensusModule, error) {
+func Create(configPath, genesisPath string, useRandomPK bool) (modules.ConsensusModule, error) {
+	cm := new(ConsensusModule)
+	c, err := cm.InitConfig(configPath)
+	if err != nil {
+		return nil, err
+	}
+	g, err := cm.InitGenesis(genesisPath)
+	if err != nil {
+		return nil, err
+	}
+	cfg := c.(*typesCons.ConsensusConfig)
```

Comment:
You can do the type assertion/casting directing in `InitConfig` and `InitiGenesis`. See this code snippet:

```go
package main

import (
	"fmt"
)

type ConfigI interface {
  GetConfigName() string  
}

var _ ConfigI = &configA{}
var _ ConfigI = &configB{}

type configA struct {}
type configB struct {}

func (c *configA) GetConfigName() string { return "A" }
func (c *configB) GetConfigName() string { return "B" }

func GetConfigA() ConfigI {
  return &configA{}
}

func GetConfigB() ConfigI {
  return &configB{}
}

func main() {
	fmt.Println("Hello, World!")
        fmt.Println(GetConfigA().GetConfigName())
       fmt.Println(GetConfigB().GetConfigName())
}
```

Reply:
Yes you can, but the argument for the creation of each module is not the `interface` it's the `structure` yet the return argument from the `InitConfig` and `InitGenesis` is the interface. This is due to the ask to add the `initConfig` and `initGenesis` to the shared interface module